File: Core/Core.EntityFramework/SharedEntity/ArkadaslikTeklif.cs
using System;

namespace Core.EntityFramework.SharedEntity
{
    public class ArkadaslikTeklifBase 
    {
        public int TeklifEdenNo { get; set; }
        public int TeklifEdilenNo { get; set; }
        public DateTime IstekTarihi { get; set; }
        public DateTime? CevapTarihi { get; set; }
        public bool? Karar { get; set; }
        public DateTime? IptalTarihi { get; set; }
        public int? IptalEdenKullaniciNo { get; set; }
        public bool? IptalEdildi { get; set; }
    }
}

File: SSB.Spa/src/app/models/arkadaslik-teklif.ts


export class ArkadaslikTeklif {
    id: string;
    arkadas: Arkadas;
    teklifEden: Arkadas;
    teklifEdilen: Arkadas;
    istekTarihi?: Date;
    cevapTarihi?: Date;
    karar?: boolean;
    iptalTarihi?: Date;
    iptalEdenKullaniciNo?: number;
    iptalEdildi?: boolean;
}

export class Arkadas {
    id: number;
    tamAdi: string;
    profilFotoUrl: string;
    sonMesajZamani: Date;
    eposta: string;
    epostaOnaylandi: boolean;
    telefonNumarasi?: string;
    telefonOnaylandi?: boolean;
    yasi: number;
}

import { SorguBase } from './sorgu-base';

export class ArkadaslikSorgusu extends SorguBase {
    constructor() {
        super();
        this.sayfa = 1;
        this.sayfaBuyuklugu = 10;
        this.kullaniciNo = true;
        this.filtreCumlesi = 'hepsi';
    }
    filtreCumlesi?: string;
    teklifEdilenler?: boolean;
    teklifEdenler?: boolean;
    kabulEdilenler?: boolean;
    cevapBeklenenler?: boolean;
    cevaplananlar?: boolean;
    silinenler?: boolean;
    kullaniciNo?: boolean;
}
